jax/_src/numpy/polynomial.py

```diff
@@ -280,16 +337,76 @@ def polyder(p: Array, m: int = 1) -> Array:
   return p[:-m] * coeff[::-1]
 
 
-_LEADING_ZEROS_DOC = """\
-Setting trim_leading_zeros=True makes the output match that of numpy.
-But prevents the function from being able to be used in compiled code.
-Due to differences in accumulation of floating point arithmetic errors, the cutoff for values to be
-considered zero may lead to inconsistent results between NumPy and JAX, and even between different
-JAX backends. The result may lead to inconsistent output shapes when trim_leading_zeros=True.
-"""
-
-@implements(np.polymul, lax_description=_LEADING_ZEROS_DOC)
 def polymul(a1: ArrayLike, a2: ArrayLike, *, trim_leading_zeros: bool = False) -> Array:
+  r"""Returns the product of two polynomials.
+
+  JAX implementation of :func:`numpy.polymul`.
+
+  Args:
+    a1: 1D array of polynomial coefficients.
+    a2: 1D array of polynomial coefficients.
+    trim_leading_zeros: If ``True`` removes the leading zeros in the return
+      value to match the result of numpy. Default is ``False``.
+
+  Returns:
+    An array of the coefficients of the product of the two polynomials.
+
+  Note:
+
+    - Setting ``trim_leading_zeros=True`` makes the output match that of numpy.
+      But prevents the function from being able to be used in compiled code. Due
+      to differences in accumulation of floating point arithmetic errors, the
+      cutoff for values to be considered zero may lead to inconsistent results
+      between NumPy and JAX, and even between different JAX backends. The result
+      may lead to inconsistent output shapes when ``trim_leading_zeros=True``.
+    - :func:`jax.numpy.polymul` differs from :func:`numpy.polymul` when an integer
+      array is given. NumPy returns the result with dtype ``int`` whereas JAX
+      returns the result with dtype ``float``.
```

Comment:
I would mention this difference in the "Returns" block: the dtype of the output is always promoted to inexact.

Reply:
Thanks! Updated.